test/e2e/base_user_test.go

```diff
@@ -0,0 +1,125 @@
+package e2e
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type baseUserIntegrationTest struct {
+	suite.Suite
+	namespace  string
+	testCtx    *framework.TestCtx
+	awaitility *wait.Awaitility
+	hostAwait  *wait.HostAwaitility
+}
+
+func (s *baseUserIntegrationTest) clearApprovalPolicyConfig() error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "toolchain-saas-config",
+		},
+	}
+
+	err := s.awaitility.KubeClient.CoreV1().ConfigMaps(s.namespace).Delete(cm.Name, nil)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *baseUserIntegrationTest) setApprovalPolicyConfig(policy string) {
+	// Create a new ConfigMap
+	cm := &corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "toolchain-saas-config",
+		},
+	}
+
+	// Clear the current approval policy
+	err := s.clearApprovalPolicyConfig()
+	require.NoError(s.T(), err)
+
+	cmValues := make(map[string]string)
+	cmValues["user-approval-policy"] = policy
+	cm.Data = cmValues
+	_, err = s.awaitility.KubeClient.CoreV1().ConfigMaps(s.namespace).Create(cm)
+	require.NoError(s.T(), err)
+
+	// Confirm it was updated
+	cm, err = s.awaitility.KubeClient.CoreV1().ConfigMaps(s.namespace).Get("toolchain-saas-config", v1.GetOptions{})
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), policy, cm.Data["user-approval-policy"])
+}
+
+func (s *baseUserIntegrationTest) createAndCheckUserSignup(specApproved bool, username string, email string,
+	conditions ...v1alpha1.Condition) (*v1alpha1.UserSignup, *v1alpha1.MasterUserRecord) {
+
+	userSignup := s.createAndCheckUserSignupNoMUR(specApproved, username, email, conditions...)
+
+	// Confirm the MUR was created
+	mur := s.assertCreatedMUR(userSignup)
```

Comment:
The `assertCreatedMUR()` function already calls `s.hostAwait.WaitForMasterUserRecord` to ensure that the MUR is created.  I could call `verifyResourcesProvisionedForSignup` however this seems like overkill, considering that `createAndCheckUserSignup` is only used indirectly by two user management tests.  Furthermore, confirming _all_ of the resources are provisioned is seemingly redundant in light of the nature of the tests which themselves are limited in their scope to just the `UserSignup` and `MasterUserRecord` resources.  I think we already have overly sufficient coverage of resource provisioning in other integration tests, wouldn't you agree?

Reply:
it's not about verifying the provisioning flow (yes, I agree the it's sufficiently covered), it's about making sure that the account is provisioned and ready before it will be banned. In other words - all resources are properly created before they will be deleted.

Reply:
Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/80/commits/47aff7e47cb17aa3ae92efe2064202c4b3837775

Reply:
Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/80/commits/944f6941d8fbf701d9ae8fdb4e4f61733aa7b8e5